src/components/transport_manager/src/bluetooth/bluetooth_connection_factory.cc

```diff
@@ -57,12 +58,21 @@ TransportAdapter::Error BluetoothConnectionFactory::CreateConnection(
   const DeviceUID& device_uid, const ApplicationHandle& app_handle) {
   LOG4CXX_TRACE(logger_, "enter. device_uid: " << &device_uid << ", app_handle: " <<
                 &app_handle);
-  BluetoothSocketConnection* connection(
-    new BluetoothSocketConnection(device_uid, app_handle, controller_));
-  TransportAdapter::Error error = connection->Start();
-  if (TransportAdapter::OK != error) {
-    LOG4CXX_ERROR(logger_, "connection::Start() failed");
-    delete connection;
+  removeFinalizedConnections();
+  TransportAdapter::Error error = TransportAdapter::ALREADY_EXISTS;
+  const ConnectionHolderIndex connection_index(device_uid, app_handle);
+  if (connections_.end() == connections_.find(connection_index))
+  {
+    ConnectionSPtr connection(
```

Comment:
Please excuse me, I was not aware of MakeShared class. Now it is fixed. See 2ed45971106b1e0212a65d65b11021d7fb9fe51